api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -203,6 +203,9 @@ type AzureManagedMachinePoolSpec struct {
 	// +kubebuilder:validation:Enum=OS;Temporary
 	// +optional
 	KubeletDiskType *KubeletDiskType `json:"kubeletDiskType,omitempty"`
+	// SubnetName specifies the Subnet where the MachinePool will be placed
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
```

Comment:
I am inclined on changing SubnetName from type `string` to `*string`. It will be easier to validate this field and perform a nil check.